src/plugins/data/common/datasets/types.ts

```diff
@@ -256,6 +258,35 @@ export interface DatasetField {
   // TODO:  osdFieldType?
 }
 
+export enum DatasetCanvasBannerComponentType {
+  Callout = 'callout',
+  Custom = 'custom',
```

Comment:
While I know OSD has made a mess of using enums but the correct convention is for enum names to be all uppercase to indicate their constant-ness.
```suggestion
  CALLOUT = 'callout',
  CUSTOM = 'custom',
```